internal/utils.go

```diff
@@ -0,0 +1,241 @@
+	}
+
+	return sortedKeys
+}
+
+// ReverseSortMapKeysByIntValue reverse sorts the keys of a map[string]int by
+// its values.
+func ReverseSortMapKeysByIntValue(imap map[string]int) []string {
+	// create keyval
+	keyval := createKeyvalFromMap(imap)
+
+	// sort the keyval in reverse order
+	keyval.sliceSortReverse()
+
+	// sort the map by its values and return the sorted keys
+	return sortKeyvalstruct(len(imap), keyval)
+}
+
+// sliceSortReverse function reverse sorts the slice.
+func (k keyvalueStructs) sliceSortReverse() {
+	sort.Slice(k, func(i, j int) bool {
+		return k[i].Value > k[j].Value
+	})
+}
+
+// SortMapKeysByMapIntValue sorts the keys of a map[string]map[string]int by a
+// the values found at a given sub value.
+func SortMapKeysByMapIntValue(imap map[string]map[string]int, criterion string) []string {
+	// create keyval
+	keyval := createKeyvalFromMapOfMap(imap, criterion)
+
+	// sort the keyval
+	keyval.sliceSort()
+
+	// sort the map by its values and return the sorted keys
+	return sortKeyvalstruct(len(imap), keyval)
+}
+
+// createKeyvalFromMapOfMap function creates a keyvaluestruct from
+// map[string]map[string]int.
+func createKeyvalFromMapOfMap(imap map[string]map[string]int, criterion string) keyvalueStructs {
+	keyval := keyvalueStructs{}
+
+	for k, v := range imap {
+		keyval = append(keyval, keyvalueStruct{k, v[criterion]})
+	}
+
+	return keyval
+}
+
+// ReverseSortMapKeysByMapIntValue reverse sorts the keys of a
+// map[string]map[string]int by a the values found at a given sub value.
+func ReverseSortMapKeysByMapIntValue(imap map[string]map[string]int, criterion string) []string {
+	// create keyval
+	keyval := createKeyvalFromMapOfMap(imap, criterion)
+
+	// sort the keyval in reverse order
+	keyval.sliceSortReverse()
+
+	// sort the map by its values and return the sorted keys
+	return sortKeyvalstruct(len(imap), keyval)
+}
+
+// DedupSortStrings removes duplicates and then sorts the given strings,
+// returning a new slice.
+func DedupSortStrings(istrings []string) []string {
+	if len(istrings) == 0 {
+		return istrings
+	}
+
+	elementmap := make(map[string]bool)
+	dedup := []string{}
+
+	for _, entry := range istrings {
+		if _, value := elementmap[entry]; !value {
+			elementmap[entry] = true
+
+			dedup = append(dedup, entry)
+		}
+	}
+
+	sort.Strings(dedup)
+
+	return dedup
+}
+
+// ProcMeminfoMBs uses gopsutil (amd64 freebsd, linux, windows, darwin, openbds
+// only!) to find the total number of MBs of memory physically installed on the
+// current system.
+func ProcMeminfoMBs() (int, error) {
+	v, err := mem.VirtualMemory()
+	if err != nil {
+		return 0, err
+	}
+
+	// convert bytes to MB
+	return convert.BytesToMB(v.Total), err
+}
+
+// PathToContent takes the path to a file and returns its contents as a string.
+// If path begins with a tilda, TildaToHome() is used to first convert the path
+// to an absolute path, in order to find the file.
+func PathToContent(path string) (string, error) {
```

Comment:
These path and file related methods seem to belong in the filepath package or similar.